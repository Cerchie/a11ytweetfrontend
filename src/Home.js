import React from 'react'

function Home() {
    return (
        <>
            <h1 className="rounded shadow-lg text-red-800 text-6xl p-6 m-6">
                Welcome to a11yTweet!
            </h1>
            <p className="rounded-lg bg-blue-50 text-red-700 text-4xl p-6 m-6">
                We've collected a list of Github repos with 'a11y' in the name.
                You can retweet them directly from our list page, or sign up to
                save your own page of links.
            </p>
            <h2 className="text-gray-600 text-2xl p-6 m-6">
                Here are some of our favorites:
            </h2>
            <ul className="rounded shadow-md bg-blue-50 text-red-700 text-2xl p-6 m-6">
                <li>
                    <a
                        href="https://github.com/Khan/tota11y"
                        className="font-bold"
                    >
                        - An accessibility visualization toolkit
                    </a>
                </li>
                <li>
                    <a
                        href="https://github.com/a11yproject/a11yproject.com"
                        className="font-bold"
                    >
                        - The A11Y Project
                    </a>{' '}
                    is a community-driven effort to make digital accessibility
                    easier.
                </li>
                <li>
                    <a
                        href="https://github.com/jsx-eslint/eslint-plugin-jsx-a11y"
                        className="font-bold"
                    >
                        - Static AST checker
                    </a>{' '}
                    for a11y rules on JSX elements.
                </li>
            </ul>
        </>
    )
}

export default Home
