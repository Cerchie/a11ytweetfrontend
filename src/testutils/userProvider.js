import React from 'react'
import UserContext from '../UserContext'

const demoUser = {
    username: 'testuser',
}

const UserProvider = ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
        {children}
    </UserContext.Provider>
)

export { UserProvider }
