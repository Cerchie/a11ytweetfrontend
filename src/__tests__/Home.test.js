import { render } from '@testing-library/react'
import React from 'react'
import Home from '../Home'
import { MemoryRouter } from 'react-router'
import { UserProvider } from '../testutils/userProvider'

test('renders comp', () => {
    render(<Home />)
})

it('matches snapshot', function () {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <Home />
            </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot()
})

it('matches snapshot when logged out', function () {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider currentUser={null}>
                <Home />
            </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot()
})
