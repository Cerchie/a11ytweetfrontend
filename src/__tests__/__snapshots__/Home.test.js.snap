// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h1
    class="rounded shadow-lg text-red-800 text-6xl p-6 m-6"
  >
    Welcome to a11yTweet!
  </h1>
  <p
    class="rounded-lg bg-blue-50 text-red-700 text-4xl p-6 m-6"
  >
    We've collected a list of Github repos with 'a11y' in the name. You can retweet them directly from our list page, or sign up to save your own page of links.
  </p>
  <h2
    class="text-gray-600 text-2xl p-6 m-6"
  >
    Here are some of our favorites:
  </h2>
  <ul
    class="rounded shadow-md bg-blue-50 text-red-700 text-2xl p-6 m-6"
  >
    <li>
      <a
        class="font-bold"
        href="https://github.com/Khan/tota11y"
      >
        - An accessibility visualization toolkit
      </a>
    </li>
    <li>
      <a
        class="font-bold"
        href="https://github.com/a11yproject/a11yproject.com"
      >
        - The A11Y Project
      </a>
       is a community-driven effort to make digital accessibility easier.
    </li>
    <li>
      <a
        class="font-bold"
        href="https://github.com/jsx-eslint/eslint-plugin-jsx-a11y"
      >
        - Static AST checker
      </a>
       for a11y rules on JSX elements.
    </li>
  </ul>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <h1
    class="rounded shadow-lg text-red-800 text-6xl p-6 m-6"
  >
    Welcome to a11yTweet!
  </h1>
  <p
    class="rounded-lg bg-blue-50 text-red-700 text-4xl p-6 m-6"
  >
    We've collected a list of Github repos with 'a11y' in the name. You can retweet them directly from our list page, or sign up to save your own page of links.
  </p>
  <h2
    class="text-gray-600 text-2xl p-6 m-6"
  >
    Here are some of our favorites:
  </h2>
  <ul
    class="rounded shadow-md bg-blue-50 text-red-700 text-2xl p-6 m-6"
  >
    <li>
      <a
        class="font-bold"
        href="https://github.com/Khan/tota11y"
      >
        - An accessibility visualization toolkit
      </a>
    </li>
    <li>
      <a
        class="font-bold"
        href="https://github.com/a11yproject/a11yproject.com"
      >
        - The A11Y Project
      </a>
       is a community-driven effort to make digital accessibility easier.
    </li>
    <li>
      <a
        class="font-bold"
        href="https://github.com/jsx-eslint/eslint-plugin-jsx-a11y"
      >
        - Static AST checker
      </a>
       for a11y rules on JSX elements.
    </li>
  </ul>
</DocumentFragment>
`;
